crowbar_framework/app/models/api/upgrade.rb

```diff
@@ -24,6 +24,10 @@ def status
       end
 
       def checks
+        upgrade_status = ::Crowbar::UpgradeStatus.new
+        # the check for current_step means to allow running the step at any point in time
+        upgrade_status.start_step if upgrade_status.current_step == "upgrade_prechecks"
```

Comment:
IMHO the design is broken, but we could fix it quite quickly here. Alternatively we could keep adding new race conditions and new cards to fix them later, but is it really a good idea to keep accumulating technical debt which will need to be fixed before the release? I'd prefer to fix them now:

1. Make the code changes which make the tests in https://github.com/aspiers/crowbar-core/pull/1 green
2. Change the interface to implement [the atomic test-and-set operation described above](https://github.com/crowbar/crowbar-core/pull/736/files#r84703091)
3. Change this calling code to use that interface.

That should all be doable in 2-3 hours IMHO.